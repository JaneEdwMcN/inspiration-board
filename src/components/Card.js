import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  render() {

    return (
      <div className="card">
        <div className="card__content">
          <p className="card__content-text">
          {this.props.text}
          { this.props.emoji && <span className="card__content-emoji">  { emoji.getUnicode(this.props.emoji) }</span>}
          </p>
        </div>
      </div>
    )
  }
}

Card.propTypes = {

};

export default Card;
