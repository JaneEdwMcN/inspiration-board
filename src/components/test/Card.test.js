import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('card', () => {
  test('that it matches an existing snapshot', () => {

    const wrapper = shallow( <Card
      id={1}
      text="Doggo"
      emoji={"heart"}
      />);

      expect(wrapper).toMatchSnapshot();
    });
  });
