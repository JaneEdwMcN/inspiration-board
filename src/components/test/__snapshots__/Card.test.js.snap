// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    >
      Doggo
      <span
        className="card__content-emoji"
      >
          
        ❤️
      </span>
    </p>
  </div>
</div>
`;
